src/sql/parse.py

```diff
@@ -217,9 +218,15 @@ def without_sql_comment(parser, line):
     return " ".join(result)
 
 
-def magic_args(magic_execute, line):
+def magic_args(magic_execute, line, cmd_from, allowed_duplicates):
+    """
+    Returns the parsed arguments from the line as parsed by magic_execute
+    """
     line = without_sql_comment(parser=magic_execute.parser, line=line)
-    return magic_execute.parser.parse_args(shlex.split(line, posix=False))
+    args = shlex.split(line, posix=False)
+    if len(args) > 1:
+        check_duplicate_arguments(magic_execute, cmd_from, args, allowed_duplicates)
```

Comment:
does `allowed_duplicates` vary with the class ? Is it not common across like `--with` etc?